### ISSUE TITLE
Packs are not listed in WEBUI for users with observer role

### ISSUE THREAD

**Issue by SURAJTHEGREAT** [StackStorm/st2#4063]
Hi Team,

In stackstorm WebUI , page **does not render** on clicking packs tab by logged in user having **observer role**.

It highlighted the below error  when checked in debug logs of browser
 `{
    **"faultstring": "User \"arul\" doesn't have required permission \"pack_search\""**
}`

How can the permission `pack_search` be appended to observer role?

Is it possible to provide `pack_search` permission for all the packs as
`---
    name: "pack_search_role"
    description: "Role which grants pack_search permission to all packs"
    permission_grants:
        -
            resource_uid: "packs"
            permission_types:
               - "pack_search"
`

Is this a know issue?

**Please look into this and advice. Also find screenshot for the same**.
![pack_search_error](https://user-images.githubusercontent.com/9215832/38351375-c0a8e8f0-38cc-11e8-866c-581e6a23d2f8.png)

Thanks,
Suraj S

**Comment by Kami** [comment]
Thanks for reporting this.

This looks like an oversight on our side - it makes sense for "observer" role to also grant `PACK_SEARCH` permission.

**Comment by SURAJTHEGREAT** [comment]
@Kami Thanks for the quick reply..

Can you validate if we can create a role that provides permission for all packs

---
    name: "pack_search_role"
    description: "Role which grants pack_search permission to all packs"
    permission_grants:
        -
            resource_uid: "packs"
            permission_types:
               - "pack_search"

**Comment by Kami** [comment]
@SURAJTHEGREAT Right now we don't support globs in `resource_uid` field and ability to grant permissions on all or a set of resources which match a particular glob.

This means you would need to explicitly grant `pack_search` permission on all the packs which are in the system.

So something like this:

```yaml
...
    -
        resource_uid: "packs:pack1"
        permission_types:
           - "pack_search"
    -
        resource_uid: "packs:pack1"
        permission_types:
           - "pack_search"
....
```

Also, #4064 has been merged and will be included in the next release (v2.7.0).

**Comment by Kami** [comment]
I opened an issue for ability to grant permission on all resources of a specific type - #4066.

This might still come handy in some scenarios even now that #4064 fix is in master.
